clients/fides-js/src/components/tcf/TcfPurposes.tsx

```diff
@@ -178,18 +178,32 @@ const TcfPurposes = ({
               ]
             : activeData.enabledPurposeIds
         }
-        onToggle={(newEnabledIds, item, toggleDetails) =>
+        onToggle={(newEnabledIds, item: RecordListItem, toggleDetails) => {
+          let filteredEnabledIds = newEnabledIds;
+          if (item.bestTranslation) {
+            // filter out tcf purpose consent since we are just dealing with custom purposes
+            filteredEnabledIds = newEnabledIds.filter(
+              (id) => !activeData.enabledPurposeIds.includes(id),
+            );
+          } else if (activeData.enabledCustomPurposeIds) {
+            /// filter out custom purpose consent since we are just dealing with TCF purposes
+            filteredEnabledIds = newEnabledIds.filter(
+              // @ts-ignore
+              (id) => !activeData.enabledCustomPurposeIds.includes(id),
```

Comment:
@eastandwestwind let's avoid ts-ignore at all cost! It's the equivalent to skipping a test just because it's failing rather than fixing the bug.
```suggestion
              (id) => !activeData.enabledCustomPurposeIds?.includes(id),
```